### PR TITLE
Fixed lp:1631954 "It is possible to create COMPRESSED columns in MyISAM tables via specifying partition engines" and lp:1671492 "Unable to add compression columns to a partitioned table via 'ALTER TABLE'" (5.7)

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/727/

File: sql/sql_table.cc [hunk 1]
```diff
@@ -11162,15 +11172,18 @@ static bool check_engine(THD *thd, const char *db_name,
...
-  List_iterator<Create_field> it(
-    const_cast<List<Create_field>&>(alter_info->create_list));
-  Create_field* sql_field= it++;
-  while (sql_field != 0 &&
-         sql_field->column_format() != COLUMN_FORMAT_TYPE_COMPRESSED)
-    sql_field= it++;
-  if (sql_field != 0 &&
-    ((*new_engine)->create_zip_dict == 0 ||
-     (*new_engine)->drop_zip_dict == 0))
+#ifdef WITH_PARTITION_STORAGE_ENGINE
```
Comment (laurynas-biveinis) [inline sql/sql_table.cc]: IIRC 5.7 removed WITH_PARTITION_STORAGE_ENGINE #ifdefs in the code

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comments as for 5.6 and a 5.7-specific one

File: sql/sql_table.cc (hunk 1 above)
Comment (percona-ysorokin) [reply sql/sql_table.cc]: good spot

**Review by robgolebiowski — APPROVED** [review]
LGTM

**Comment by percona-ysorokin** [comment]
Updated Jenkins build link
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/730/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**